server/src/storage/compressed_data_manager.rs

```diff
@@ -170,36 +176,99 @@ impl CompressedDataManager {
         Ok(())
     }
 
+    /// Return an [`ObjectMeta`] for a compressed file in `query_data_folder` that belongs to the
+    /// column at `column_index` in the table with `table_name` and contains all of the compressed
+    /// segments within the given range of time and value. If some compressed data that belongs to
+    /// `column_index` in `table_name` is still in memory, save it to disk first. If no files belong
+    /// to the column at `column_index` for the table with `table_name` an empty [`Vec`] is
+    /// returned, while a [`DataRetrievalError`](ModelarDbError::DataRetrievalError) is returned if:
+    /// * A table with `table_name` does not exist.
+    /// * A column with `column_index` does not exist.
+    /// * The compressed files could not be listed.
+    /// * The end time is before the start time.
+    /// * The max value is smaller than the min value.
+    #[allow(clippy::too_many_arguments)]
+    pub(super) async fn save_merge_and_get_saved_compressed_files(
+        &mut self,
+        table_name: &str,
+        column_index: u16,
+        start_time: Option<Timestamp>,
+        end_time: Option<Timestamp>,
+        min_value: Option<Value>,
+        max_value: Option<Value>,
+        query_data_folder: &Arc<dyn ObjectStore>,
+    ) -> Result<Vec<ObjectMeta>, ModelarDbError> {
+        // Retrieve object_metas that represent the relevant files for table_name and column_index.
+        let relevant_apache_parquet_files = self
+            .save_and_get_saved_compressed_files(
+                table_name,
+                column_index,
+                start_time,
+                end_time,
+                min_value,
+                max_value,
+                query_data_folder,
+            )
+            .await?;
+
+        // Compact the compressed Apache Parquet files if multiple are returned to ensure order.
```

Comment:
The [merging](https://www.infoworld.com/article/3685496/compactor-a-hidden-engine-of-database-performance.html) of multiple smaller files is commonly known as compaction (see, e.q., [Apache Cassandra](https://cassandra.apache.org/doc/latest/cassandra/operating/compaction/index.html)), however since this is the only location `compact` is used instead of `merge` I have replaced it with `merge` for consistency.